src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,101 +1,108 @@
-import React, { useEffect, useState } from 'react';
-import './PostDetails.scss';
-
-import { Loader } from '../Loader';
+import React, { useState, useEffect } from 'react';
 import { NewCommentForm } from '../NewCommentForm';
-import { getPostDetails } from '../../api/posts';
-import { removePostComment, getPostComments } from '../../api/comments';
+import './PostDetails.scss';
+import { Post, Comment } from '../../react-app-env';
+import { getComments, delComment } from '../../api/comments';
+import { getPostbyId } from '../../api/posts';
 
 type Props = {
-  postId: number,
+  postId: number | undefined;
 };
 
-export const PostDetails: React.FC<Props> = ({ postId }) => {
+export const PostDetails: React.FC<Props> = ({
+  postId,
+}) => {
+  const [commentsList, setCommentsList] = useState<Comment[] | undefined>([]);
```

Comment:
```suggestion
  const [commentsList, setCommentsList] = useState<Comment[] | undefined>(undefined);
```
in case with `[ ]` it never been undefined